CK3toEU4/Source/EU4World/EU4World.cpp

```diff
@@ -668,6 +668,18 @@ std::optional<std::pair<std::string, std::shared_ptr<CK3::Title>>> EU4::World::d
 			maxDev = provinceWeight;
 		}
 	}
+
+	// If the title's name is transfering to EU4, make sure it makes sense. Thrace should use Constantinople's name...
+	if (toReturn.second->isRenamed())
+	{
+		// ... if they belong to the same owner, Constantinople also has a custom name and they are in the same mapping
+		const auto& myDuchyCapital = toReturn.second->getDJLiege()->second->getCapital().second;
+		if (myDuchyCapital->isRenamed() && ck3Titles.contains(myDuchyCapital->getName()) &&
+			 myDuchyCapital->getDFLiege()->first == toReturn.second->getDFLiege()->first)
+		{
+			toReturn.second->overrideDisplayName(myDuchyCapital->getDisplayName());
+		}
+	}
```

Comment:
In addition to the comment from @Zemurin, if you place this all in a small function you can use guard clauses for the checks and avoid deep nesting: https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html